alt-tab-macos/ui/Application.swift

```diff
@@ -93,4 +102,22 @@ class Application: NSApplication, NSApplicationDelegate, NSWindowDelegate {
         hideUi()
         DispatchQueue.global(qos: .userInteractive).async { window?.focus() }
     }
+
+    func closeSelectedWindow(_ window: TrackedWindow?) {
+        DispatchQueue.global(qos: .userInteractive).async { window?.close() }
+        isOutdated = true
+        // slight delay needed to give the window time to close before updating the thumbnails
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .milliseconds(75), execute: {
```

Comment:
I'm afraid this is too fragile. Many windows will take longer to close so it will fail. I think there are way to listen to windows being closed and application being quit. That would be one way to follow the OS work completion.

Another way would be to optimistically update our UI to remove the thumbnail(s), and consider that the OS doing the work is an async task now that will happen eventually (which it should). We could either remove the thumbnails, or mark them somehow like opacity reduced, black, red cross, etc.

I think these are the 2 ways to go: (a) listen to window close events somehow, or (b) update UI optimistically and let the OS deal with execution of the order

Reply:
Yeah, as I said, im not happy with the delays either.

I'll start with your second suggestion, as this is easily ruled out. Many applications ask you, if you really want to close the window. iTerm with multiple Tabs for example. In this case I think it would be bad UX if we change our UI when the application doesn't.

Now to the first suggestion, which I also thought about. I think this would definitely be the way to go. I briefly looked into [AXObserver](https://developer.apple.com/documentation/applicationservices/axobserver) with which we should be able to do this. Unfortunately I haven't got the notifications working. I'll take another look at this solution since this would be the best.

Reply:
Update: I got the `AXObserver` working. I'm now looking into how I will implement it to be as flexible as possible. Who knows for what it's good in the future :)

Reply:
I'm now using the `AXObserver`. So far it seems to work beautifully. I even got closing fullscreen windows to work with it :)
But extensive testing is more than welcome 🚀